lib/backend/client/hostendpoint.go

```diff
@@ -0,0 +1,52 @@
+package client
+
+import (
+	. "github.com/projectcalico/libcalico/lib/backend/objects"
+)
+
+// HostEndpointInterface has methods to work with Service resources.
+type HostEndpointInterface interface {
+	List(hostname, name *string) ([]HostEndpoint, error)
+	Get(hostname, name string) (*HostEndpoint, error)
+	Create(hostname, name string, data *HostEndpoint) error
+	Update(hostname, name string, data *HostEndpoint) error
+	Delete(hostname, name string) error
+}
+
+// services implements ServicesNamespacer interface
+type hostEndpoints struct {
+	r  *Client
+}
+
+// newServices returns a services
+func newHostEndpoints(c *Client) *hostEndpoints {
+	return &hostEndpoints{c}
+}
+
+// List takes a Metadata, and returns the list of hot endpoints that match that Metadata
+// (wildcarding mising fields)
+func (h *hostEndpoints) List(hostname, name *string) ([]HostEndpoint, error) {
```

Comment:
Is using a pointer for an optional parameter a Go idiom?  In the stdlib I've seen

- multiple functions with different names, e.g. in the regexp package, you've got Find*
- using varargs to let you specify only what you want
- using an options struct to let you specify what you want